main.js

```diff
@@ -75,24 +76,83 @@ function createTray() {
 }
 
 function startApp() {
-  const modalPath = `file://${__dirname}/process.html`
+  const modalPath = `file://${__dirname}/process.html` // TODO about:blank?
   backgroundWindow = new BrowserWindow({ show: false });
   backgroundWindow.loadURL(modalPath)
   createTray();
 
-  // 2018-09-12 mp: Disabling check for updates until updater talks to omnitruck
-  // so that users don't get message around missing app-update.yml on App startup.
-  //
-  // Get the menuItem so that the updater can toggle it's state. There's probably
-  // a way to encapsulate this better.
-  // backgroundWindow.once('ready-to-show', () => {
-  //   updater.checkForUpdates(trayMenu.items[0]);
-  // })
+  app.emit('do-update-check');
+  console.log("update check emission complete!")
 }
 
 function quitApp() {
   backgroundWindow = null;
   app.quit();
 }
 
+mixlibInstallUpdater.on('start-update-check', () => {
+  console.log("got start-update-check") ;
+  // disable the menu to prevent concurrent checks
+  trayMenu.getMenuItemById('updateCheck').enabled  = false;
+});
+
+mixlibInstallUpdater.on('update-not-available', () => {
+  console.log("got update-not-available");
+  // If they picked the menu option, show a notification dialog.
+  if (requestFromMenu) {
+    const noUpdateDialog = require('./src/no_update_dialog.js');
+    noUpdateDialog.open();
+  }
+  WSTray.instance().displayNotification(false);
+});
+
+mixlibInstallUpdater.on('update-available', (updateInfo) => {
+  console.log("got update-available: " + util.inspect(updateInfo));
+  // Only display the notification. Changing the menu text is a lot of work
+  // and will be done in the next re-factor.
+  //  TODO - why ? This ... *looked* like we could just update the label...
```

Comment:
https://github.com/hovancik/stretchly/blob/master/app/main.js#L510

One of the other tray app / windows based applications I have looked at uses a `getMenuTray()` method and calls `setContextMenu` any time they need to update the menu. I think Jon's idea about pushing this logic into a class that can manage existing state and be responsible for the updates makes sense.